sync/sync_test.go

```diff
@@ -47,19 +47,30 @@ func TestSyncSimpleRequestingHead(t *testing.T) {
 	err = syncer.SyncWait(ctx)
 	require.NoError(t, err)
 
-	exp, err := remoteStore.Head(ctx)
-	require.NoError(t, err)
+	// force sync to update underlying stores.
+	syncer.wantSync()
 
-	have, err := localStore.Head(ctx)
-	require.NoError(t, err)
-	assert.Equal(t, exp.Height(), have.Height())
-	assert.Empty(t, syncer.pending.Head())
+	// we need to wait for a flush
+	assert.Eventually(t, func() bool {
+		exp, err := remoteStore.Head(ctx)
+		require.NoError(t, err)
 
-	state := syncer.State()
-	assert.Equal(t, uint64(exp.Height()), state.Height)
-	assert.Equal(t, uint64(2), state.FromHeight)
-	assert.Equal(t, uint64(exp.Height()), state.ToHeight)
-	assert.True(t, state.Finished(), state)
+		have, err := localStore.Head(ctx)
+		require.NoError(t, err)
+
+		state := syncer.State()
+
+		ok := true
+		ok = ok && exp.Height() == have.Height()
+		ok = ok && syncer.pending.Head() == nil
+
+		ok = ok && uint64(exp.Height()) == state.Height
+		ok = ok && uint64(2) == state.FromHeight
+		ok = ok && uint64(exp.Height()) == state.ToHeight
+		ok = ok && state.Finished()
```

Comment:
Looks unusual. Can it be just couple of require.Equal() calls?

Reply:
Agree, but if you will take a look at definition of `assert.Eventually` you will see that we need `bool` as a result.

That's why `assert/require.Equal` here doesn't make sense.

Reply:
Alternative can be to fail fast with `if`. Suggestion:
```
if exp.Height() != have.Height() {
  return false
}
```

Reply:
Fixed! Indeed, I have repeated this pattern from C/Java times (😢 ) but Go's `switch` works well here.